pkg/logql/syntax/ast.go

```diff
@@ -1170,7 +1170,9 @@ const (
 	// internal expressions not represented in LogQL. These are used to
 	// evaluate expressions differently resulting in intermediate formats
 	// that are not consumable by LogQL clients but are used for sharding.
-	OpRangeTypeQuantileSketch = "__quantile_sketch_over_time__"
+	OpRangeTypeQuantileSketch     = "__quantile_sketch_over_time__"
+	OpRangeTypeFirstWithTimestamp = "__first_over_time_ts__"
```

Comment:
AFAIU, this is used to represent that both `first_over_time` and `last_over_time` is shardable with downstream engine. why not call it `__first_over_time_sharded__` instead?. Failing to understand the purpose of `_ts__` suffix here.

Reply:
It's for the queriers to tell them to include the original timestamp in the result and not override it with the timestamp of the step.